consensus/blake3pow/consensus.go

```diff
@@ -297,19 +297,22 @@ func (blake3pow *Blake3pow) verifyHeader(chain consensus.ChainHeaderReader, head
 		if header.GasUsed() > header.GasLimit() {
 			return fmt.Errorf("invalid gasUsed: have %d, gasLimit %d", header.GasUsed(), header.GasLimit())
 		}
-		// Verify the block's gas usage and (if applicable) verify the base fee.
-		if !chain.Config().IsLondon(header.Number()) {
-			// Verify BaseFee not present before EIP-1559 fork.
-			if header.BaseFee() != nil {
-				return fmt.Errorf("invalid baseFee before fork: have %d, expected 'nil'", header.BaseFee())
-			}
-			if err := misc.VerifyGaslimit(parent.GasLimit(), header.GasLimit()); err != nil {
-				return err
-			}
-		} else if err := misc.VerifyEip1559Header(chain.Config(), parent, header); err != nil {
-			// Verify the header's EIP-1559 attributes.
+		// Verify the block's gas usage and verify the base fee.
+		// Verify that the gas limit remains within allowed bounds
+		parentGasLimit := parent.GasLimit()
+		if err := misc.VerifyGaslimit(parentGasLimit, header.GasLimit()); err != nil {
 			return err
 		}
+		// Verify the header is not malformed
+		if header.BaseFee() == nil {
```

Comment:
in `block.go` we have a method to validate headers. We should just update that method to enforce `BaseFee() != nil` like it already does for all the other fields of the header, and then we don't need to use `BaseFee() == nil` checks like this